analytic_diffuse/models.py

```diff
@@ -15,16 +17,54 @@ def checkinput(func):
     def wrapper(*args, **kwargs):
         # Check validity of input shapes
         phi = args[0]
+        # Check whether input was scalar.
+        scalar = np.isscalar(args[1])
+
         theta = args[1]
-        if not phi.shape == theta.shape:
+
+        if not scalar:
+            theta = np.array(theta)
+            if phi is not None:
+                phi = np.array(phi)
+
+        if scalar and theta > np.pi/2:
+            return 0
+
+        if phi is not None and not scalar and phi.shape != theta.shape:
             raise ValueError(
                 "phi and theta must have the same shape: {}, {}".format(
                     str(phi.shape), str(theta.shape)
                 )
             )
-        result = func(*args, **kwargs)
+        result = func(phi, theta, *args[2:], **kwargs)
+
         # Set pixels outside the horizon to zero.
-        result[theta > np.pi / 2] *= 0
+        if not scalar:
+            result[theta > np.pi / 2] = 0
+
+        # If a scalar was passed in, return a scalar.
+        return result
+    return wrapper
+
+
+def projected(func):
+    """Enforce input shapes and horizon cutoff."""
```

Comment:
This docstring should be on the other wrapper.

Reply:
Yeah, I just copied it and forgot about it. I've fixed it now.